internal/yml/find_node_test.go

```diff
@@ -1,7 +1,6 @@
-package utils_test
+package yml
```

Comment:
Why do you use the same package? For these types of tests I feel its better convention to test as a "consumer", eg without visibility into the package internals.

Reply:
I see what you mean.  I've always written tests from the outside (ie: "black box testing") in other languages (Java, C#...) but the way I learned in Go (rather common, but maybe not systematic) is to put tests within same package as code under test (ie: "white box testing").  What I really like about that approach is that it fairly easy to directly test critical internal components that would otherwise only be tested indirectly, and have much more numerous test cases for those internal components (which might be tricker if tested externally/indirectly).  Anyway, I think both approaches are quite valid and there's probably no reason to just have one or the other.

Reply:
> What I really like about that approach is that it fairly easy to directly test critical internal components that would otherwise only be tested indirectly, 

I had a go mentor once tell me that if a critical component can't be tested with blackbox testing, its either not critical, or your code is shit xD 

> Anyway, I think both approaches are quite valid and there's probably no reason to just have one or the other.

The reason I usually opt for blackbox is it forces you to ensure that the component behaves in the ways its designed to be used, which by consequence allows you to write tests that document your code / provide examples of how it's intended to be used 

If theres no reason to white box test here, then I'd suggest we stick with the black box pattern and if we run into a situation where white box testing is needed we can revisit the discussion with that context.

Reply:
OK, agreed, I moved most tests to their own package, there was only one case `getIndentSize()` that was an internal implementation detail that I did not want to expose outside the package, but still wanted to test directly.  If we can agree that we will use black-box in most cases, but allow white-box when it makes sense, then I'm good with it! 👍